### PR TITLE
use indexed payment query

### PR DESCRIPTION
**Pull request by mshick**
### Test Plan (Steps to test):

1. Login, go to purchases page
1. Notice how quickly purchases show up! Rejoice!
1. Make a new purchase... There may be a delay in it showing on the list. But it should within 1 minute.

### CODE REVIEW

**Comment by mshick** [comment]
@allanlasser Let me know if the delay really bothers you. We could move to a secondary fetch for the most recent item and if it differs push it to the top of the list... Could be a weird UX though.

File: components/payments.tsx [hunk 1]
```diff
@@ -139,31 +127,20 @@ function getLineItems(invoiceOrSession: Stripe_Invoice | Stripe_CheckoutSession)
...
-        <Box sx={{ textAlign: 'right' }}>
-          {invoice?.invoice_pdf ? (
-            <Link title="Download Receipt" target="_blank" rel="noreferrer" href={invoice?.invoice_pdf}>
-              <IconButton color="text">
-                <BsReceipt size={16} />
-              </IconButton>
-            </Link>
-          ) : null}
-          {invoice?.paid ? <BsCheckCircleFill title="Paid" color="green" size={16} /> : null}
-        </Box>
```
Comment (allanlasser) [inline components/payments.tsx:166]: Losing the link to the invoice PDF feels like a downgrade. Any way to keep this functionality?

**Review by allanlasser — CHANGES_REQUESTED** [review]
I do think the need to fully refresh the page in order to see the new purchase in the list isn't optimal, especially during a demo flow. I think a way around this is to use Apollo to [refetch the purchase list on a timed interval or on a "refresh button" click](https://egghead.io/lessons/react-refetch-data-with-the-apollo-query-component-either-manually-or-on-timed-intervals)? [React Query](https://react-query.tanstack.com/reference/useQuery#_top) supports a similar concept with its `refetchInterval` option.

**Comment by allanlasser** [comment]
That being said, the loading speed improvement is a huge step forward!

File: components/payments.tsx (hunk 1 above)
Comment (mshick) [reply components/payments.tsx:166]: The only way I could get to a reasonable depth was to just get invoice items and exclude the invoice. The indexing set up is pretty fragile right now.

We only have invoices for a subset of purchases  — subscriptions — so it seemed like an acceptable loss.

**Comment by mshick** [comment]
Might be good to refetch in general, though once we have `get` requests for Stripe indexes working the update should be almost instantaneous.

**Review by allanlasser — APPROVED** [review]
Feels way better to have one loading spinner than 3, and was able to see my purchase added to the list about 15s after purchase.

**Pull request merged**